contrib/windows-cmake/CMakeLists.txt

```diff
@@ -3,48 +3,105 @@ project(hwloc
     LANGUAGES C
     VERSION 2.7.0)
 
-# Available configuration options
-# HWLOC_SKIP_LSTOPO doesn't build/install lstopo
-# HWLOC_SKIP_TOOLS doesn't build/install other hwloc tools
-# HWLOC_SKIP_INCLUDES doesn't install headers
+enable_testing()
 
-set(TOPDIR ../..)
+option(HWLOC_ENABLE_TESTING "Enable testing" ON)
+option(HWLOC_SKIP_LSTOPO "don't build/install lstopo")
+option(HWLOC_SKIP_TOOLS "don't build/install other hwloc tools")
+option(HWLOC_SKIP_INCLUDES "don't install headers")
 
-# Configure based on the MSVC config
+set(TOPDIR ${PROJECT_SOURCE_DIR}/../..)
+
+set(CMAKE_STATIC_LIBRARY_PREFIX lib)
+set(CMAKE_SHARED_LIBRARY_PREFIX lib)
+# "libhwloc.*" naming for MSVC and non-MSVC
 
 configure_file(${TOPDIR}/contrib/windows/hwloc_config.h include/hwloc/autogen/config.h COPYONLY)
-configure_file(${TOPDIR}/contrib/windows/static-components.h include/static-components.h COPYONLY)
-configure_file(${TOPDIR}/contrib/windows/private_config.h include/private/autogen/config.h COPYONLY)
 
-file(READ ${CMAKE_CURRENT_BINARY_DIR}/include/private/autogen/config.h PRIVATE_CONFIG_H)
-string(REPLACE "#define SIZEOF_VOID_P 8"        "#define SIZEOF_VOID_P ${CMAKE_SIZEOF_VOID_P}"        PRIVATE_CONFIG_H "${PRIVATE_CONFIG_H}")
+# Configure dynamically based on platform capabilities
+include(CheckIncludeFile)
+include(CheckSymbolExists)
+include(CheckCSourceCompiles)
+
+check_include_file("dirent.h" HAVE_DIRENT_H)
+check_include_file("unistd.h" HAVE_UNISTD_H)
+check_include_file("malloc.h" HAVE_MALLOC_H)
+check_include_file("memory.h" HAVE_MEMORY_H)
+
+check_symbol_exists(mkstemp "stdlib.h" HAVE_MKSTEMP)
+check_symbol_exists(memalign "malloc.h" HAVE_MEMALIGN)
+
+check_symbol_exists(strncasecmp "strings.h" HAVE_STRNCASECMP)
```

Comment:
a bit concerned about these checks. Does this try a compile? If yes it might give wrong results with `/Oi` on MSVC. (Depends on how the symbol is checked for. If it is a dummy function call with "wrong" arguments it will raise an error with /Oi due to the wrong arguments.)

Reply:
with or without CMAKE_C_FLAGS=/Oi the visual studio results are the same here. The errors in either case are like `error C2065: 'memalign': undeclared identifier`

```
-- Looking for dirent.h
-- Looking for dirent.h - not found
-- Looking for unistd.h
-- Looking for unistd.h - not found
-- Looking for malloc.h
-- Looking for malloc.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for mkstemp
-- Looking for mkstemp - not found
-- Looking for memalign
-- Looking for memalign - not found
-- Looking for strncasecmp
-- Looking for strncasecmp - not found
```